functionary/core/models/environment.py

```diff
@@ -29,3 +29,23 @@ class Meta:
 
     def __str__(self):
         return f"{self.team.name} - {self.name}"
+
+    def variables(self, variable_names=None):
```

Comment:
Will add the docstring. 

This is used in the next step to get all environment variables visible from the environment. Optionally trimmed to just the values you pass in. Since the environment hides some from the Team, this seems like the best place to put the variable accessor.

Reply:
I think having a helper that gets you the full set of variables for the environment (its own, plus those inherited from the team) makes sense.  The part where you can pass in a list of variables to filter that down to feels a little out of scope for a model helper function to me.

And in any case, I think if you have this return a set answering that question can just be done via the set intersection.  For instance:

```python
In [14]: my_vars = {"two", "four"} # this is what you were proposing gets passed in as variable_names
In [15]: env_vars = {"one", "two", "three", "four"} # let's say this is what would have come back from calling variables()
In [16]: my_vars & env_vars
Out[16]: {'four', 'two'}
```

Reply:
This function returns the variables, not just the names. You could do a comprehension when you get them all back but it seems silly to do that when you can just pass in the ones you want. User has similar methods for permissions.